### PR TITLE
Dimensionality Reduction

### PR DESCRIPTION
**Pull request by nicriv831**

### CODE REVIEW

File: dimensionality-reduction/dimensionality-reduction.py [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+scaler = StandardScaler()
+x_train_scaled = scaler.fit_transform(X_train)
+
+# %%
+# make model, pipeline and fit pipeline
+
+knn_nopca = KNeighborsClassifier()
+knn_nopca_pipe = make_pipeline(scaler, knn_nopca)
+knn_nopca_pipe.fit(x_train_scaled, y_train)
+
```
Comment (Caellwyn) [inline dimensionality-reduction/dimensionality-reduction.py:69]: You already scaled your data once.  You don't need to scale it both in the pipeline and before the pipeline.

File: dimensionality-reduction/dimensionality-reduction.py [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+# %%
+# score 25 component pipeline
+
+scores.loc['pca25_knn', 'test_accuracy'] = pca_25_pipe.score(X_test, y_test)
+scores.loc['pca25_knn', 'train_accuracy'] = pca_25_pipe.score(X_train, y_train)
+scores
+
+# %%
+# Try with 50 componets
+
```
Comment (Caellwyn) [inline dimensionality-reduction/dimensionality-reduction.py:129]: If you are going to tune your PCA for accuracy, it's more efficient to use a loop rather than trying different values by hand.

**Review by Caellwyn — CHANGES_REQUESTED** [review]
Please fix your double-scaling issue and resubmit.

**Review by Caellwyn — APPROVED** [review]
